mlip_testing/analysis/surfaces/S24/analyse_S24.py

```diff
@@ -0,0 +1,188 @@
+"""Analyse S24 benchmark."""
+
+from __future__ import annotations
+
+from pathlib import Path
+
+from ase.io import read, write
+import pytest
+
+from mlip_testing.analysis.utils.decorators import build_table, plot_parity
+from mlip_testing.analysis.utils.utils import mae
+from mlip_testing.calcs.models.models import MODELS
+
+CALC_PATH = (
+    Path(__file__).parent.parent.parent.parent
+    / "calcs"
+    / "surfaces"
+    / "S24"
+    / "outputs"
+)
+OUT_PATH = (
+    Path(__file__).parent.parent.parent.parent / "app" / "data" / "surfaces" / "S24"
+)
+
+
+def compute_adsorption_energy(
+    surface_e: float, mol_surf_e: float, molecule_e: float
+) -> float:
+    """
+    Compute adsorption energy.
+
+    Parameters
+    ----------
+    surface_e
+        Energy of the clean surface.
+    mol_surf_e
+        Energy of the molecule+surface system.
+    molecule_e
+        Energy of the isolated molecule.
+
+    Returns
+    -------
+    float
+        Adsorption energy.
+    """
+    return mol_surf_e - (surface_e + molecule_e)
+
+
+def system_names() -> list:
+    """
+    Get list of system names.
+
+    Returns
+    -------
+    list
+        List of all system names.
+    """
+    system_names = []
+    for model_name in MODELS:
+        for system_path in (CALC_PATH / model_name).glob("*.xyz"):
+            if system_path.name == "s24_mol_surface_atoms.xyz":
+                continue
+            system_names.append(system_path.stem)
+        break
+    return system_names
+
+
+@pytest.fixture
+@plot_parity(
+    filename=OUT_PATH / "figure_adsorption_energies.json",
+    title="Adsorption energies",
+    x_label="Predicted adsorption energy / eV",
+    y_label="Reference adsorption energy / eV",
+    hoverdata={
+        "System": system_names(),
+    },
+)
+def adsorption_energies() -> dict[str, list]:
+    """
+    Get adsorption energies for all systems.
+
+    Returns
+    -------
+    dict[str, list]
+        Dictionary of all reference and predicted adsorption energies.
+    """
+    results = {"ref": []} | {mlip: [] for mlip in MODELS}
+    ref_stored = False
+
+    for model_name in MODELS:
+        for i, system_path in enumerate(sorted((CALC_PATH / model_name).glob("*.xyz"))):
```

Comment:
```suggestion
        for system_path in sorted((CALC_PATH / model_name).glob("*.xyz")):
```
I don't think `i` is used so it (and the enumerate) can probably be removed?

Sorting is a good idea though, we should probably do this everywhere, as the glob order isnt' guaranteed.